src/Application.php

```diff
@@ -201,13 +201,13 @@ public function pipe($path, $middleware = null)
     {
         // Lazy-load middleware from the container when possible
         $container = $this->container;
-        if (null === $middleware && is_string($path) && $container && $container->has($path, true)) {
+        if (null === $middleware && is_string($path) && $container && $container->has($path)) {
```

Comment:
Noting for posterity: The boolean `true` argument existed because an earlier revision of zend-servicemanager for v3 defined it; when provided, it indicated whether or not to look in abstract factories when performing the check. We have since removed that flag in zend-servicemanager's develop branch as it violates the LSP, but also because it was leading to many subtle errors (it was far too easy to forget to add it, and have a false negative lookup).